ideascube/library/tests/test_migrations.py

```diff
@@ -20,6 +20,126 @@
+        subtitle='Volume 1364: Migrations')
+    book.save()
+    BookSpecimen(serial='3456', book_id=book.id).save()
+    BookSpecimen(serial='4567', book_id=book.id).save()
+
+    old_item = StockItem(name='testing django apps')
+    old_item.save()
+    old_specimen = Specimen(barcode='3456', item_id=old_item.id)
+    old_specimen.save()
+
+    # Keep for later
+    old_item_id = old_item.id
+    old_specimen_id = old_specimen.id
+
+    migration.run_migration()
+
+    Book = migration.new_apps.get_model('library', 'Book')
+    BookSpecimen = migration.new_apps.get_model('library', 'BookSpecimen')
+    StockItem = migration.new_apps.get_model('monitoring', 'StockItem')
+    Specimen = migration.new_apps.get_model('monitoring', 'Specimen')
+
+    assert Book.objects.count() == 3
+    assert BookSpecimen.objects.count() == 4
+    assert StockItem.objects.count() == 3
+    assert Specimen.objects.count() == 4
+
+    books = Book.objects.order_by('name')
+    bookspecimens = BookSpecimen.objects.order_by('item__name', 'barcode')
+    items = StockItem.objects.order_by('name')
+    specimens = Specimen.objects.order_by('item__name', 'barcode')
+
+    book = books[0]
+    assert book.name == 'A book which was not in the stock'
+    assert book.module == 'library'
+
+    item = items[0]
+    assert item.book.id == book.id
+
+    bookspecimen = bookspecimens[0]
+    assert bookspecimen.barcode == '1234'
+    assert bookspecimen.serial is None
+    assert bookspecimen.item.book.id == book.id
+
+    specimen = specimens[0]
+    assert specimen.bookspecimen.id == bookspecimen.id
+    assert specimen.item.id == item.id
+
+    bookspecimen = bookspecimens[1]
+    assert bookspecimen.barcode == '2345'
+    assert bookspecimen.serial is None
+    assert bookspecimen.item.book.id == book.id
+
+    specimen = specimens[1]
+    assert specimen.bookspecimen.id == bookspecimen.id
+    assert specimen.item.id == item.id
+
+    book = books[1]
+    assert book.name == 'Les blagues à Toto'
+    assert book.module == 'library'
+    assert book.specimens.count() == 0
+
+    item = items[1]
+    assert item.book.id == book.id
+    assert item.specimens.count() == 0
+
+    book = books[2]
+    assert book.id == old_item_id
+    assert book.name == 'Testing Django Applications'
+    assert book.subtitle == 'Volume 1364: Migrations'
+    assert book.module == 'library'
+
+    item = items[2]
+    assert item.id == old_item_id
+    assert item.book.id == book.id
+
+    bookspecimen = bookspecimens[2]
+    assert bookspecimen.id == old_specimen_id
+    assert bookspecimen.barcode == '3456'
+    assert bookspecimen.serial is None
+    assert bookspecimen.item.book.id == book.id
+
+    specimen = specimens[2]
+    assert specimen.id == old_specimen_id
+    assert specimen.bookspecimen.id == bookspecimen.id
+    assert specimen.item.id == item.id
+
+    bookspecimen = bookspecimens[3]
+    assert bookspecimen.barcode == '4567'
```

Comment:
Good idea, especially with regards to https://github.com/ideascube/ideascube/pull/642#discussion_r91484847